scipy/linalg/flapack.pyf.src

```diff
@@ -887,6 +887,48 @@ interface
 
    end subroutine <prefix>getrf
 
+
+   subroutine <prefix2>sytrd(lower,n,a,lda,d,e,tau,work,lwork,info)
+     ! Reduce a real symmetric matrix A to real symmetric
+     ! tridiagonal form T by an orthogonal similarity transformation:
+     ! Q**T * A * Q = T.
+
+     callstatement (*f2py_func)((lower?"L":"U"),&n,a,&lda,d,e,tau,work,&lwork,&info);
+     callprotoargument char*,int*,<ctype2>*,int*,<ctype2>*,<ctype2>*,<ctype2>*,<ctype2>*,int*,int*
+
+     integer intent(hide),depend(a) :: lda=MAX(shape(a,0),1)
+     integer intent(hide),depend(a) :: n=shape(a,1)
+     integer optional,intent(in),check(lower==0||lower==1) :: lower = 0
+
+     <ftype2> dimension(lda,n),intent(in,out,copy,out=c) :: a
+     <ftype2> dimension(n),intent(out),depend(n) :: d
+     <ftype2> dimension(n-1),intent(out),depend(n) :: e
+     <ftype2> dimension(n-1),intent(out),depend(n) :: tau
+     integer intent(in),optional,depend(n),check(lwork>=MAX(n,1)) :: lwork = MAX(n,1)
```

Comment:
Lwork is not needed to be `n` or higher. It just needs to be greater than or equal to 1. For this you can keep the default but the check needs to be as 

    integer intent(in),optional,depend(n),check(lwork>=1||lwork==-1) :: lwork = MAX(n,1)

Reply:
Here we already define `sytrd_lwork` for -1 query, but some people do it manually so better to keep it general.